### PR TITLE
4444 Move dispensary reports

### LINKED ISSUE #4444: Move Program reports to report page
## Is your feature request related to a problem? Please describe 👀
Move Program reports to report page

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #4444

# 👩🏻‍💻 What does this PR do?
Move program reports to dispensary

# 🧪 Testing

- [ ] Have `uses program module` store pref on
- [ ] Go to reports
- [ ] Should see list of reports
- [ ] Reports are correct

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

**Review by mark-prins — CHANGES_REQUESTED** [review]
woo! thanks Roxy! great to see the reports come across 😁 
I jumped in to review, as Andrei is out for a few days, hope that's ok.

ah! this old chestnut. 

Screenshot 2024-07-24 at 11 34 50 AM

There's two instances of imports from 'packages/' in the codebase atm:

`ReportArgumentModal.tsx` needs to use
`import { useAuthContext } from '@openmsupply-client/common';`

and `reports/src/DetailView/DetailView.tsx` needs
`import { JsonData } from '@openmsupply-client/programs';`

is this the point to mention styles? Not sure, as I'm jumping in here to do a review as Andrei is away..

The current style isn't matching what's in figma:

Screenshot 2024-07-24 at 11 46 13 AM

This is my [suggested change branch](https://github.com/msupply-foundation/open-msupply/compare/4444-Move-Program-reports-to-report-page...4444-Move-Program-reports-to-report-page(suggestion)) which looks closer:

Screenshot 2024-07-24 at 12 05 09 PM

File: client/packages/reports/src/ListView/ListView.tsx [hunk 1]
```diff
@@ -63,6 +70,16 @@ export const ListView = () => {
           reportWithArgs={reportWithArgs}
           setReportWithArgs={setReportWithArgs}
         />
+        {store?.preferences?.omProgramModule && (
```
Comment (mark-prins) [inline client/packages/reports/src/ListView/ListView.tsx:73]: maybe also check that there are some reports? Otherwise there's a blank widget shown 🤷

Comment (roxy-dao) [reply client/packages/reports/src/ListView/ListView.tsx:74]: I've added a condition on all widgets to hide if no reports 😉

File: client/packages/system/src/Report/api/hooks/useReport.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { REPORT } from './keys';
 export const useReport = (id: string) => {
   const { reportApi, storeId } = useReportGraphQL();
 
-  const queryKey = [REPORT, storeId];
+  const queryKey = [REPORT, storeId, id];
```
Comment (roxy-dao) [inline client/packages/system/src/Report/api/hooks/useReport.ts:9]: 👀  fix for the bug... cache wasn't getting cleared so reports needed double clicks to get the right one

File: client/packages/reports/src/ListView/ListView.tsx (hunk 1 above)
Comment (roxy-dao) [reply client/packages/reports/src/ListView/ListView.tsx:74]: Also if you wanna test it with Stock status reports~  https://github.com/msupply-foundation/open-msupply-reports/pull/71

**Review by mark-prins — APPROVED** [review]
❤️  looks good - thanks Roxy!

**Pull request merged**